web/packages/teleport/src/services/api/api.ts

```diff
@@ -181,13 +191,10 @@ const api = {
 
     let mfaResponseForRetry;
     try {
-      const challenge = await auth.getMfaChallenge({
-        scope: MfaChallengeScope.ADMIN_ACTION,
-      });
-      mfaResponseForRetry = await auth.getMfaChallengeResponse(challenge);
+      mfaResponseForRetry = await auth.getAdminActionMfaResponse();
```

Comment:
There's a circular dependency here. `auth` depends on `api` and `api` depends on `auth` (I know that's not a new thing). Let me know if this doesn't make sense, but maybe it could be `api` responsibility to take care of MFA? 
I can imagine that instead of manually calling `auth.getAdminActionMfaResponse(true)` in every place where we need the MFA response, we would do: 
```ts
    return api
      .post(cfg.getAwsRdsDbsDeployServicesUrl(integrationName), req, null, {
        scope: 'admin-action',
        reusable: true
      })
```
And then `api` would call the mfa context.

Reply:
In some cases we need to reuse the same MFA response for multiple API calls, hence the reusable response:

```
const mfaResponse = await auth.getAdminActionMfaResponse(true);
    return ctx.userService
      .createUser(u, ExcludeUserField.Traits, mfaResponse)
      .then(result => setUsers([result, ...users]))
      .then(() =>
        ctx.userService.createResetPasswordToken(u.name, 'invite', mfaResponse)
      );
```

We could include an `mfaContext` in both `auth` and `api` instead?